pkg/util/dynamichelper/dynamichelper.go

```diff
@@ -92,6 +135,83 @@ func (dh *dynamicHelper) Ensure(ctx context.Context, objs ...kruntime.Object) er
 	return nil
 }
 
+func (dh *dynamicHelper) ensureUnstructuredObj(ctx context.Context, o *UnstructuredObj) error {
+	gvr, err := dh.resolve(o.obj.GroupVersionKind().GroupKind().String(), o.obj.GroupVersionKind().Version)
+	if err != nil {
+		return err
+	}
+
+	create := false
+	obj, err := dh.dynamicClient.Resource(*gvr).Namespace(o.obj.GetNamespace()).Get(ctx, o.obj.GetName(), metav1.GetOptions{})
+	if err != nil {
+		if !notFound(err) {
+			return err
+		}
+		create = true
+	}
+	if create {
+		dh.log.Printf("Create %s", keyFunc(o.GroupVersionKind().GroupKind(), o.GetNamespace(), o.GetName()))
+		if _, err = dh.dynamicClient.Resource(*gvr).Namespace(o.obj.GetNamespace()).Create(ctx, &o.obj, metav1.CreateOptions{}); err != nil {
+			return err
+		}
+		return nil
```

Comment:
```suggestion
		return dh.dynamicClient.Resource(*gvr).Namespace(o.obj.GetNamespace()).Create(ctx, &o.obj, metav1.CreateOptions{})
```

Reply:
your suggested code does not even compile as the func only one return value:
```
too many arguments to return
	have (*unstructured.Unstructured, error)
	want (error)
```

Reply:
ah, yes indeed, missed that